t/app/controller/auth_profile.t

```diff
@@ -0,0 +1,202 @@
+    $mech->content_contains( 'Now check your email', "found check your email" );
+    $link = $mech->get_link_from_email;
+    $mech->log_out_ok;
+    $mech->get_ok($link);
+    isnt $mech->uri->path, '/auth/change_email/success', "not redirected to the change_email page";
+    $mech->content_contains('Sorry');
+};
+
+my $test_phone_bad = '01214960000000';
+my $test_landline = '01214960000';
+my $test_mobile = '+61491570156';
+
+subtest "Test add/verify/change phone page" => sub {
+    # Still signed in from the above test
+    $mech->get_ok('/auth');
+    $mech->submit_form_ok({
+        with_fields => {
+            username => $test_email,
+            password_sign_in => $test_password,
+        },
+    });
+
+    $mech->follow_link_ok({url => '/auth/change_phone'});
+    $mech->submit_form_ok( { with_fields => { username => "" } }, "submit blank change phone form" );
+    is $mech->uri->path, '/my', 'redirected';
+    $mech->content_contains('successfully removed');
+
+    $mech->follow_link_ok({url => '/auth/change_phone'});
+    $mech->submit_form_ok({ with_fields => { username => $test_phone_bad } });
+    $mech->content_contains( 'Please check your phone number is correct', "found expected error" );
+
+    FixMyStreet::override_config({
+        SMS_AUTHENTICATION => 1,
+        PHONE_COUNTRY => 'GB',
+    }, sub {
+        $mech->submit_form_ok({ with_fields => { username => $test_landline } });
+    });
+    is $mech->uri->path, '/my', 'redirected';
+    $mech->content_contains('successfully added');
+
+    FixMyStreet::override_config({
+        SMS_AUTHENTICATION => 1,
+        PHONE_COUNTRY => 'GB',
+    }, sub {
+        $mech->follow_link_ok({url => '/auth/verify/phone'});
+        $mech->submit_form_ok({ with_fields => { username => $test_landline } });
+    });
+    $mech->content_contains( 'Please enter a mobile number', "found expected error" );
+
+    FixMyStreet::override_config({
+        SMS_AUTHENTICATION => 1,
+        TWILIO_ACCOUNT_SID => 'AC123',
+    }, sub {
+        $mech->submit_form_ok({ with_fields => { username => $test_mobile } });
+    });
+    is $mech->uri->path, '/auth/verify/phone', "still on change phone page";
+    $mech->content_contains( 'Now check your phone', "found check your phone" );
+
+    $mech->submit_form_ok({
+        with_fields => { code => '00000' }
+    }, 'submit incorrect code');
+    $mech->content_contains('Try again');
+
+    my $code = $twilio->get_text_code;
+    $mech->submit_form_ok({
+        with_fields => { code => $code }
+    }, 'submit correct code');
+
+    my $user = FixMyStreet::App->model('DB::User')->find( { phone => $test_mobile } );
+    ok $user, "user exists";
+    is $user->email_verified, 1;
+    is $user->email, $test_email, 'email still same';
+    is $mech->uri->path, '/my', "redirected to /my page";
+    $mech->content_contains('successfully verified');
+    $mech->logged_in_ok;
+};
```

Comment:
Not sure if here is the correct place but it might be useful to have some tests to check verifying a phone number for an email verified account merges in reports etc for a phone verified account with the same name. And I guess vice versa.

Reply:
Yeah, and same in admin `user_edit` looking red currently on https://codecov.io/gh/mysociety/fixmystreet/pull/1856/src/perllib/FixMyStreet/App/Controller/Admin.pm#L1401 – think I have time so will add something there too.